### DISCUSSION TITLE
I had a .swc folder

### DISCUSSION THREAD

**Discussion by maoxiaoke** [swc-project/swc#4023] [Q&A] [answered]
When I run `swc.transformSync`，a `.swc` directory was generated in the root. My questions are: 

+ what is this used for
+ How can i eliminate it when transforming

**Comment by kdy1** [comment, marked as answer]
> what is this used for

This is used for bytecode cache of plugins.

> How can i eliminate it when transforming

Update swc. It's already fixed

**Reply by ywang1724** [reply]
Appeared again with v1.3.59

**Comment by brynshanahan** [comment]
I'm also getting the .swc in every monorepo package that swc is used. Is there a config option I can parse that will make all monorepo packages use a .swc folder in the root directory? @kdy1

**Reply by kwonoj** [reply]
https://github.com/swc-project/swc/blob/61dc611da3842b2cf1dfe7e4b9a5d06f172e679c/node-swc/src/types.ts#L654

**Reply by brynshanahan** [reply]
Thank you 🙏

**Reply by brynshanahan** [reply]
Hey @kwonoj I have a .swrc config file in the root of my monorepo is there any way to make all the cache files use the same directory?

ie using the below config creates the cache directory in every package's node_modules directory and I can't use `../../node_modules` because the package is not guaranteed to be at the same level
```
experimental: {
  "cacheRoot": "node_modules/.cache/swc"
}
```

I tried
```
experimental: {
  "cacheRoot": "/node_modules/.cache/swc"
}
```
But I can't see the folder when using that. I guess I'm asking if there's any way to make paths relative to the directory

```
experimental: {
  "cacheRoot": "<configDir>/node_modules/.cache/swc"
}
```

**Reply by kwonoj** [reply]
In my opinion you shouldn't use node_modules for the cache location. Cache is generally better to be a canonical location to known tmp.

And no, we do not have a plan to let config to dyncaminally calculate via such syntax yet. It is a entirely different problem we are thinking separately.

**Reply by brynshanahan** [reply]
I thought the js community had standardized on using the node_modules/.cache for most caches. I believe that's what TurboRepo and quite a few other tools do, it's helpful for us because node_modules are often cached between pipeline jobs. 

Are there any downsides you can see using that node_modules/.cache directory?

**Reply by kwonoj** [reply]
My recommendation is scoped to the swc only, not for the other tools. I am aware some tools use node_modules for the cache location but so far it is not a standard and also it depends on the each tools usecase.
